extensions/amp-mowplayer/0.1/amp-mowplayer.js

```diff
@@ -255,20 +286,237 @@ class AmpMowplayer extends AMP.BaseElement {
   }
 
   /**
-   * Sends 'listening' message to the Mowplayer iframe to listen for events.
+   * Disable sticky player
    * @private
    */
-  listenToFrame_() {
-    if (!this.iframe_) {
-      return;
+  onStickyDisable_() {
+    this.stickyDisableWorker_.call();
+  }
+
+  /**
+   * Set player height on resize
+   * @private
+   * @param {Object} data
+   */
+  onResize_(data) {
+    if (!window.mowStickyEnabled) {
+      this.attemptChangeHeight(data.state.dimensions.height + 10);
+    }
+  }
+
+  /**
+   * Set full width player on mobile when enabled from platform
+   * @private
+   * @param {Object} data
+   */
+  onReady_(data) {
+    if (
+      /(iPhone|iPad|iPod|Android|webOS|BlackBerry|Windows Phone)/gi.test(
+        navigator.userAgent
+      ) &&
+      data.state.config.mobile_full_width &&
+      !this.mofileFullApplied_
+    ) {
+      this.mofileFullApplied_ = true;
+      const repaint = () => {
+        const rect = this.element.getBoundingClientRect();
+        if (rect.width < window.screen.availWidth) {
+          const leftMargin = rect.left;
+          const rightMargin = window.screen.availWidth - rect.right;
+          setStyle(this.element, 'marginLeft', '-' + leftMargin + 'px');
+          setStyle(this.element, 'marginRight', '-' + rightMargin + 'px');
+        }
```

Comment:
Please **don't** change sizing/positioning styles dynamically. **This causes content layout shift.**

Reply:
@alanorozco Removed